token/program-2022/src/extension/default_account_state/instruction.rs

```diff
@@ -0,0 +1,115 @@
+use {
+    crate::{
+        check_program_account, error::TokenError, instruction::TokenInstruction,
+        state::AccountState,
+    },
+    num_enum::{IntoPrimitive, TryFromPrimitive},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        program_error::ProgramError,
+        pubkey::Pubkey,
+    },
+    std::convert::TryFrom,
+};
+
+/// Default Account State extension instructions
+#[derive(Clone, Copy, Debug, PartialEq, IntoPrimitive, TryFromPrimitive)]
+#[repr(u8)]
+pub enum DefaultAccountStateInstruction {
+    /// Initialize a new mint with the default state for new Accounts.
+    ///
+    /// Fails if the mint has already been initialized, so must be called before
+    /// `InitializeMint`.
+    ///
+    /// The mint must have exactly enough space allocated for the base mint (82
+    /// bytes), plus 83 bytes of padding, 1 byte reserved for the account type,
+    /// then space required for this extension, plus any others.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[writable]` The mint to initialize.
+    Initialize,
+    /// Update the default state for new Accounts. Only supported for mints that include the
+    /// `DefaultAccountState` extension.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   * Single authority
+    ///   0. `[writable]` The mint.
+    ///   1. `[signer]` The mint freeze authority.
```

Comment:
What if we added a DefaultAccountState authority instead?  That could still be set as the freeze authority if the user wants.  But this is an expensive extension (data wise) already so what's another 32bytes

Reply:
Not sure I'm following why it's expensive, data-wise. The Mint extension is only extension-type (1) + length (2) + AccountState (1) = 4 bytes.
I did consider a separate authority, but adding 32 bytes seemed like too much.

Reply:
jfc, my brain swapped `AccountState` for `state::Account` during my previous reading!

Reply:
Ohhhh 😅  your comment makes perfect sense, then!

Reply:
... given that clarification, how is this looking to you @mvines ?